core/src/main/java/org/elasticsearch/transport/RequestHandlerRegistry.java

```diff
@@ -63,4 +79,66 @@ public boolean isForceExecution() {
     public String getExecutor() {
         return executor;
     }
+
+    @Override
+    public String toString() {
+        return handler.toString();
+    }
+
+    private static class TransportChannelWrapper implements TransportChannel {
```

Comment:
Can you make a `DelegatingTransportChannel` which just blindly delegates to another and then overwrite that with the one that unregisters the task? It'd make it more obvious what this is doing.